packages/webapp/src/components/TileDashboard/useDynamicTileVisibility.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useState, useLayoutEffect, RefObject } from 'react';
+import type { TypeCountTile } from './index';
+
+// Source: https://github.com/que-etc/resize-observer-polyfill
+interface ResizeObserver {
+  observe(target: Element): void;
+  unobserve(target: Element): void;
+  disconnect(): void;
+}
+
+interface useDynamicTileVisibilityParams {
+  containerRef: RefObject<HTMLElement>;
+  typeCountTiles: TypeCountTile[];
+  gap?: number;
+  tileWidth?: number;
+  moreButtonWidth?: number;
+  minWidthDesktop?: number;
+  rowsPerView?: {
+    desktop: number;
+    mobile: number;
+  };
+}
+
+export const useDynamicTileVisibility = ({
+  containerRef,
+  typeCountTiles,
+  gap = 4,
+  tileWidth = 90,
+  moreButtonWidth = 90,
+  minWidthDesktop = 600,
+  rowsPerView = { desktop: 1, mobile: 2 },
+}: useDynamicTileVisibilityParams) => {
+  const [visibleIconTiles, setVisibleIconTiles] = useState<TypeCountTile[]>([]);
+  const [hiddenIconTiles, setHiddenIconTiles] = useState<TypeCountTile[]>([]);
+
+  const totalTiles = typeCountTiles.length;
```

Comment:
I think this can be moved to `useLayoutEffect`, and `isDesktopView`can be `const`.

Reply:
Is there an advantage to calculating `totalTiles` from within the `useLayoutEffect`?

Reply:
If it's in the `useLayoutEffect`, I think it will be calculated only once when the component is rendered.